cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -151,6 +152,52 @@ impl<State> CoreEnvironment<State> {
         Ok(store_path)
     }
 
+    /// Creates a [ContainerBuilder] from the environment.
+    ///
+    /// The sink is typically a [File](std::fs::File), [Stdout](std::io::Stdout)
+    /// but can be any type that implements [Write](std::io::Write).
+    ///
+    /// While container _images_ can be created on any platform,
+    /// only linux _containers_ can be run with `docker` or `podman`.
+    /// Building an environment for linux on a non-linux platform (macos),
+    /// will likely fail unless all packages in the environment can be substituted.
+    ///
+    /// There are mitigations for this, such as building within a VM or container.
+    /// Such solutions are out of scope at this point.
+    /// Until then, this function will error with [CoreEnvironmentError::ContainerizeUnsupportedSystem]
+    /// if the environment is not linux.
+    ///
+    /// [Self::lock]s if necessary.
+    ///
+    /// Technically this does write to disk as a side effect (i.e. by locking).
+    /// It's included in the [ReadOnly] struct for ergonomic reasons
+    /// and because it doesn't modify the manifest.
+    ///
+    /// todo: should we always write the lockfile to disk?
+    pub fn build_container(
+        &mut self,
+        flox: &Flox,
+    ) -> Result<ContainerBuilder, CoreEnvironmentError> {
+        if std::env::consts::OS != "linux" {
```

Comment:
note nonblocking: it would be nice to be able to set `--system` explicitly when hacking on this on a mac, because I have remote aarch64-linux builders setup